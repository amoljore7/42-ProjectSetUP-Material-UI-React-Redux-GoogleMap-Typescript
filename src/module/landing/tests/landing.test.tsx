import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

import LandingPage from '../component/landing';

const routeComponentPropsMock = {
  history: {} as any,
  location: {} as any,
  match: {} as any
};

test('loads and displays landing page', async () => {
  const { asFragment } = render(<LandingPage {...routeComponentPropsMock} />);
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('landingWrapper').children.length).toEqual(5);

  //   const signInButton = screen.getByText('Sign In');
  //   fireEvent.click(signInButton);
  //   await waitFor(() => expect(signInButton).not.toBeInTheDocument());
  //   expect(signInButton).not.toBeInTheDocument();
});
