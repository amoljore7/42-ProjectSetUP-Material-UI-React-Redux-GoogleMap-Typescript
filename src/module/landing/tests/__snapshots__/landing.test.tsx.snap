// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays landing page 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-wrapperStyle-3"
      data-testid="landingWrapper"
    >
      <h1
        class="makeStyles-headingStyle-1"
      >
        HEALTHFUL
      </h1>
      <h3
        class="makeStyles-headings-2"
      >
        You are accessing PHI
      </h3>
      <h5
        class="makeStyles-headings-2"
      >
        By clicking ’Sign in’ you agree that you will only use protected health information 
      </h5>
      <h5
        class="makeStyles-headings-2"
      >
        (PHI) to provide, coordinate or manage healthcare treatment for this patient
      </h5>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-signIn-5"
        style="background-color: rgb(68, 145, 224); color: rgb(255, 255, 255);"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Sign In
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </main>
</DocumentFragment>
`;
